discord/integrations.py

```diff
@@ -0,0 +1,199 @@
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+import datetime
+from collections import namedtuple
+from .utils import _get_as_snowflake, get, parse_time
+from .user import User
+from .errors import InvalidArgument
+from .enums import try_enum, ExpireGracePeriod, ExpireBehaviour
+
+class IntegrationAccount(namedtuple('IntegrationAccount', 'id name')):
+    """Represents an integration account.
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The account ID.
+    name: :class:`str`
+        The account name.
+    """
+
+    __slots__ = ()
+
+    def __repr__(self):
+        return '<IntegrationAccount id={0.id} name={0.name!r}>'.format(self)
+
+class Integration:
+    """Represents a guild integration.
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The integration ID.
+    name: :class:`str`
+        The integration name.
+    guild: :class:`Guild`
+        The guild of the integration.
+    type: :class:`str`
+        The integration type (i.e. Twitch).
+    enabled: :class:`bool`
+        Whether the integration is currently enabled.
+    syncing: :class:`bool`
+        Where the integration is currently syncing.
+    role: :class:`Role`
+        The role which the integration uses for subscribers.
+    expire_behaviour: :class:`ExpireBehaviour`
+        The behaviour of expiring subscribers.
+    expire_grace_period: :class:`ExpireGracePeriod`
+        The grace period for expiring subscribers.
+    user: :class:`User`
+        The user for the integration.
+    account: :class:`IntegrationAccount`
+        The integration account information.
+    synced_at: :class:`datetime.datetime`
+        When the integration was last synced.
+    """
+
+    __slots__ = ('id', '_state', 'guild', 'name', 'enabled', 'type',
+                 'syncing', 'role', 'expire_behaviour', 'expire_behavior',
+                 'expire_grace_period', 'synced_at', 'user', 'account')
+
+    def __init__(self, guild, data):
+        self.guild = guild
+        self._state = guild._state
+        self._from_data(data)
+
+    def __repr__(self):
+        return '<Integration id={0.id} name={0.name!r} type={0.type!r}>'.format(self)
+
+    def _from_data(self, integ):
+        self.id = _get_as_snowflake(integ, 'id')
+        self.name = integ['name']
+        self.type = integ['type']
+        self.enabled = integ['enabled']
+        self.syncing = integ['syncing']
+        self.role = get(self.guild.roles, id=int(integ['role_id']))
```

Comment:
I suggest storing the `role_id` and then resolving it later to prevent strong references.